haystack/nodes/retriever/link_content.py

```diff
@@ -0,0 +1,197 @@
+import logging
+from datetime import datetime
+from http import HTTPStatus
+from typing import Optional, Dict, List, Union, Callable
+from urllib.parse import urlparse
+
+import requests
+from boilerpy3 import extractors
+from requests import Response
+from requests.exceptions import InvalidURL
+
+from haystack import Document, __version__
+from haystack.document_stores import BaseDocumentStore
+from haystack.nodes import PreProcessor
+from haystack.nodes.retriever.base import BaseRetriever
+from haystack.schema import FilterType
+
+logger = logging.getLogger(__name__)
+
+
+def html_content_handler(response: Response) -> Optional[str]:
+    """
+    Extracts content from the response text using the boilerpy3 extractor.
+    """
+    extractor = extractors.ArticleExtractor(raise_on_failure=False)
+    try:
+        extracted_content = extractor.get_content(response.text)
+        return extracted_content
+    except Exception:
+        return None
+
+
+def pdf_content_handler(response: Response) -> Optional[str]:
+    # TODO: implement this
+    return None
+
+
+class LinkContentRetriever(BaseRetriever):
+    """
+    LinkContentRetriever fetches content from a URL and converts it into a list of Document objects.
+
+    LinkContentRetriever supports the following content types:
+    - HTML
+
+    """
+
+    def __init__(self, pre_processor: Optional[PreProcessor] = None):
+        """
+        Creates a LinkContentRetriever instance.
+        :param pre_processor: PreProcessor to apply to the extracted text
+        """
+        super().__init__()
+        self.pre_processor = pre_processor
+        self.content_handlers: Dict[str, Callable] = {"html": html_content_handler, "pdf": pdf_content_handler}
+
+    def __call__(self, url: str, timeout: int = 3, doc_kwargs: Optional[dict] = None) -> List[Document]:
+        """
+        Fetches content from a URL and converts it into a list of Document objects.
+        :param url: URL to fetch content from.
+        :param timeout: Timeout in seconds for the request.
+        :param doc_kwargs: Optional kwargs to pass to the Document constructor.
+        :return: List of Document objects.
+        """
+        if not url or not self._is_valid_url(url):
+            raise InvalidURL("Invalid or missing URL: {}".format(url))
+
+        doc_kwargs = doc_kwargs or {}
+        extracted_doc = {"url": url}
+
+        response = self._get_response(url, timeout)
+        if not response:
+            return []
+
+        # will handle non-HTML content types soon, add content type resolution here
+        handler = "html"
+        if handler in self.content_handlers:
+            extracted_content = self.content_handlers[handler](response)
+            if extracted_content:
+                extracted_doc.update({"text": extracted_content})
```

Comment:
```suggestion
                extracted_doc.update({"content": extracted_content})
```
You could add the text to the `content` field, so that there is no need for subsequent conversions.
Am I missing something?

Reply:
No, I just carried over old code without rethinking this part.